modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/FutureDatesFunction.java

```diff
@@ -38,9 +40,13 @@ public Boolean apply(Object object1, Object object2) {
 		LocalDate localDate = DateParameterUtil.getLocalDate(
 			object1.toString());
 
-		if (localDate.isBefore(
-				DateParameterUtil.getLocalDate(object2.toString()))) {
+		SimpleDateFormat formatter = new SimpleDateFormat(
+			"yyyy-MM-dd HH:mm:ss.S");
+
+		LocalDate currentDate = DateParameterUtil.getLocalDate(
+			formatter.format(object2));
```

Comment:
hey @ericthmoritsuka 

As discussed with @carolmariaabb, could you please change the logic to some like this:

```
if(object2 instanceof Date) {
   formatter.format(object2)
}
else  {
  //keep the same logic
}
```

Not sure, but maybe this logic don't broke the actual logic.

Thank you

Reply:
I see, it makes sense. I generalized the entire code because I didn't take into consideration that object2 might be some other thing. I will work on it. Thanks!